### ISSUE TITLE
`401` errors from Nerdgraph

### ISSUE THREAD

**Issue by aswanson-nr** [newrelic/newrelic-quickstarts#777]
## Description

The validation workflow that verifies quickstarts via the nerdgraph API is intermittently returning a `401` error. 

## Steps to Reproduce

- https://github.com/newrelic/newrelic-quickstarts/runs/5022458251?check_suite_focus=true

## Expected Behavior

- [ ] Success/failure of validation is reported.

**Comment by aswanson-nr** [comment]
Update: Errors appear to be coming from nerdgraph. Working with the unified team to figure out the issue

**Comment by nr-kkenney** [comment]
i heard mention of throttling / handling errors, and thought this might be another opportunity to try out [cockatiel](https://github.com/connor4312/cockatiel) (or a similar solution).

I had an example usage PR from before: https://github.com/newrelic/docs-website/pull/3219/files.

tl;dr -- In the simplest case, we can retry requests & implement exponential backoff with jitter for reach request very easily. If it's the case that we simply make too many requests (and we shouldn't make more), there are also other built in patterns that we can utilize like circuit breaker. 

Not sure if there is a strict rate limiting / throttling functionality, but I assume we still want some retry functionality in place.

**Comment by aswanson-nr** [comment]
> i heard mention of throttling / handling errors, and thought this might be another opportunity to try out [cockatiel](https://github.com/connor4312/cockatiel) (or a similar solution).
> 
> I had an example usage PR from before: https://github.com/newrelic/docs-website/pull/3219/files.
> 
> tl;dr -- In the simplest case, we can retry requests & implement exponential backoff with jitter for each request very easily. If it's the case that we simply make too many requests (and we shouldn't make more), there are also other built in patterns that we can utilize like circuit breaker.
> 
> Not sure if there is a strict rate limiting / throttling functionality, but I assume we still want some retry functionality in place.
@nr-kkenney Just FYI, I made a ticket specifically for the rate limit issue #783 and added your comment.

**Comment by aswanson-nr** [comment]
## The problem
Too many concurrent requests to Nerdgraph are causing issues on larger pull requests (changes to 80+ quickstarts/install plans.)

### Possible resolutions
- Transport batching 
  - Still constrained by the nerdgraph timeout
  - Small batch sizes could work well
  - **This is the plan**
- Submit mutation can accept an array of quickstarts
  - Would require a solid amount of work from Devex
- Remove concurrency, let the script run the requests in sequence
  - Could be a good stopgap until one of the other solutions goes in, or it might be just fine to leave.
  - Could just run 5 at a time
  - Do this if transport batching doesn’t work out
- Retry functionality in addition to this stuff?
  - Yes, we want to in addition to other changes

### Resolution AC
- [ ] Quickstart/install plan mutation requests are batched using transport batching, with a batch size of 5 
  - [ ] If transport batching doesn't work, limit concurrent requests to 5
- [ ] Quickstart/install plan mutation requests are retried with a backoff
